fml/raster_thread_merger.h

```diff
@@ -34,20 +35,33 @@ class RasterThreadMerger
   //
   // If the task queues are the same, we consider them statically merged.
   // When task queues are statically merged this method becomes no-op.
-  void MergeWithLease(size_t lease_term);
-
-  // Un-merges the threads now, and resets the lease term to 0.
+  void MergeWithLease(int lease_term);
+
+  // Gets the shared merger from current merger object
+  fml::RefPtr<fml::SharedThreadMerger> GetSharedRasterThreadMerger() const;
+
+  // Creates a new merger from parent, share the inside shared_merger member
+  // when the platform_queue_id and raster_queue_id are same.
+  static fml::RefPtr<fml::RasterThreadMerger> CreateOrShareThreadMerger(
+      const fml::RefPtr<fml::RasterThreadMerger>& parent_merger,
+      TaskQueueId platform_id,
+      TaskQueueId raster_id);
+
+  // Un-merges the threads now if current caller is the last merge caller,
+  // and it resets the lease term to 0, otherwise it will remove the caller
+  // record and return. The multiple caller records were recorded after
+  // |MergeWithLease| or |ExtendLeaseTo| method.
   //
   // Must be executed on the raster task runner.
   //
   // If the task queues are the same, we consider them statically merged.
   // When task queues are statically merged, we never unmerge them and
   // this method becomes no-op.
-  void UnMergeNow();
+  void UnMergeNowIfLastOne();
 
   // If the task queues are the same, we consider them statically merged.
   // When task queues are statically merged this method becomes no-op.
-  void ExtendLeaseTo(size_t lease_term);
+  void ExtendLeaseTo(int lease_term);
```

Comment:
Why a switch to an int?  I guess technically someone can use negative numbers now.

Reply:
Done. I forgot it that previous lease_term is a atomic_int because there's a `const int RasterThreadMerger::kLeaseNotSet = -1;`. Now I replace all `int` to `size_t`  and all `atomic_int` to `atomic_size_t`